Rename classes prefixed with Apigility

weierophinney opened the pull request:
There are three classes/interfaces prefixed with the term "Apigility":

- In zf-apigility-admin, ApigilityVersionController
- In zf-apigility, ApigilityModuleInterface
- In zf-apigility-provider, ApigilityProviderInterface

This patch updates the maps to map these to versions using the prefix "ApiTools".

weierophinney committed:
fix: rename classes prefixed with Apigility

There are three classes/interfaces prefixed with the term "Apigility":

- In zf-apigility-admin, ApigilityVersionController
- In zf-apigility, ApigilityModuleInterface
- In zf-apigility-provider, ApigilityProviderInterface

This patch updates the maps to map these to versions using the prefix
"ApiTools".

michalbundyra committed:
Replace part of class names for all new projects

Expressive <-> Mezzio
Apigility <-> ApiTools
Zend <-> Laminas

michalbundyra committed:
Fix and add tests to cover previous changes (renaming classes to new projects)

michalbundyra commented on src/Autoloader.php:
```diff
@@ -94,7 +94,12 @@ private static function createPrependAutoloader(array $namespaces, ClassLoader $
+                    . strtr(substr($class, strlen($check)), [
+                        'ApiTools' => 'Apigility',
+                        'Mezzio' => 'Expressive',
+                        'Laminas' => 'Zend',
```
we need unit tests for all these, as I've tried to comment it out each of above line any all tests still were fine

michalbundyra commented on src/Autoloader.php:
```diff
@@ -131,7 +136,12 @@ private static function createAppendAutoloader(array $namespaces, ArrayObject $l
+                . strtr(substr($class, strlen($check)), [
+                    'Apigility' => 'ApiTools',
+                    'Expressive' => 'Mezzio',
+                    'Zend' => 'Laminas',
```
we do have now all above covered in test

michalbundyra commented on test/classes.php:
```diff
@@ -49,7 +53,7 @@ class ProblemDetails {}
 
 namespace Laminas {
-    class Expressive {}
+    class Mezzio {}
```
this has been changed, as after migration we do not have Expressive any more

weierophinney committed:
qa: additional tests around classes named using known namespaces

This patch adds test cases around the following:

- Zend\Expressive\Hal\LinkGenerator\ExpressiveLinkGenerator
- ZF\Apigility\ApigilityModuleInterface
- ZF\Apigility\Provider\ApigilityProviderInterface
- ZF\Apigility\Admin\Controller\ApigilityVersionController

Each of these contains a known namespace in the class or interface name,
and we need to ensure that the autoloader aliases to a renamed variant.

weierophinney committed:
qa: disable CS checks for long lines in Apigility section of provider

michalbundyra committed:
qa: add tests to cover "reverse autoloader" changes
